lib/ex_aws/kms_client_encryption.ex

```diff
@@ -0,0 +1,52 @@
+defmodule ExAws.KMS.ClientEncryption.Result do
+  @enforce_keys [:encrypted_data_key, :ciphertext]
+  defstruct [:encrypted_data_key, :ciphertext]
+end
+
+defmodule ExAws.KMS.ClientEncryption do
+
+  @doc "Encrypts a map as json with a data key generated for the master key_id provided"
+  @spec encrypt(data :: Map.t(), key_id :: String.t()) :: ExAws.KMS.ClientEncryption.Result
+  def encrypt_map(%{} = data, key_id) do
+    config = ExAws.Config.new(:kms)
+    json = config[:json_codec].encode!(data)
+    encrypt(json, key_id)
+  end
+
+  @doc "Decrypts the result of a call to encrypt_map() (Note: keys will be strings)"
+  @spec decrypt(message :: ExAws.KMS.ClientEncryption.Result) :: Map.t()
+  def decrypt_map(message) do
+    plaintext = decrypt(message)
+    config = ExAws.Config.new(:kms)
+    {:ok, obj} = config[:json_codec].decode(plaintext)
+    obj
+  end
+
+  @doc "Encrypts plaintext with a data key generated for the master key_id provided"
+  @spec encrypt(plaintext :: String.t(), key_id :: String.t()) :: ExAws.KMS.ClientEncryption.Result
+  def encrypt(plaintext, key_id) do
+    case  ExAws.KMS.generate_data_key(key_id, [{:key_spec, "AES_256"}]) |> ExAws.request do
+      {:ok, %{"CiphertextBlob" => encrypted_data_key, "KeyId" => _, "Plaintext" => data_key}} ->
```

Comment:
If you're only going to handle the `{:ok` case then you should just use `ExAws.request!` and this function should be `def encrypt!`. The same goes for the `decrypt` function.

Ideally I think you take a different approach that looks like the following:
```
key_id
|> ExAws.KMS.generate_data_key(key_spec: "AES_256")
|> ExAws.request!
|> ExAws.KMS.ClientEncryption.encrypt(plain_text)
```

One of the guiding principles behind ExAws is the separation of pure logic from actual requests to AWS. This module in its current state violates those principles because calling these functions has hard coded requests to AWS. This makes it increasingly difficult to support supplying optional config to ExAws.request!, as well as choosing between the `!` and non `!` versions.

By making the ClientEncryption stuff just pure logic, we give all that power back to the users.